### PR TITLE
chore(sdk): `SealedBlock` generic over header and body

### PR DESCRIPTION
**Pull request by emhane**
Makes `SealedBlock` type definition generic over header and body type, which is a necessary first step to use `SealedBlock` in `Block` trait method signatures

### CODE REVIEW

File: crates/primitives-traits/src/header/sealed.rs [hunk 1]
```diff
@@ -12,24 +12,48 @@ use serde::{Deserialize, Serialize};
...
     hash: BlockHash,
     /// Locked Header fields.
     #[as_ref]
     #[deref]
-    header: Header,
+    header: H,
 }
 
-impl SealedHeader {
+impl<H: Compact> Compact for SealedHeader<H> {
```
[the remark this answers is not in the document]
Comment (emhane) [reply crates/primitives-traits/src/header/sealed.rs:26]: do you want to push some code to the branch? :D @joshieDo

[the remark this answers is not in the document]
Comment (Rjected) [reply crates/primitives-traits/src/header/sealed.rs:26]: We can remove this if we change the `Compact` derive macro to support generics, right? And we probably should if we're going to be using a a `Sealed<T>` type for more things

[the remark this answers is not in the document]
Comment (Rjected) [reply crates/primitives-traits/src/header/sealed.rs:26]: Ah that PR makes sense, this is a better solution, especially given `Sealed*` types are usually not put in the DB. So let's definitely avoid adding generics to the macro for now

File: crates/primitives/src/block.rs [hunk 2]
```diff
@@ -299,16 +299,25 @@ impl BlockWithSenders {
 /// Withdrawals can be optionally included at the end of the RLP encoded message.
 #[cfg_attr(any(test, feature = "arbitrary"), derive(arbitrary::Arbitrary))]
 #[cfg_attr(any(test, feature = "reth-codec"), reth_codecs::add_arbitrary_tests(rlp, 32))]
-#[derive(Debug, Clone, PartialEq, Eq, Default, Serialize, Deserialize, Deref, DerefMut)]
-pub struct SealedBlock {
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize, Deref, DerefMut)]
+pub struct SealedBlock<H = Header, B = BlockBody> {
```
Comment (mattsse) [inline crates/primitives/src/block.rs:303]: having to generics here is likely not convenient, ideally this should use the Block trait

**Comment by mattsse** [comment]
please link the other prs, it's easy to miss

**Comment by emhane** [comment]
end goal is to move to `alloy_primitives::Sealed`. hence pr just adds this to the type def, to unblock other prs.

**Comment by Rjected** [comment]
> to unblock other prs

Is this blocking any PRs or issues right now? If so, what's being blocked?

**Comment by emhane** [comment]
> > to unblock other prs
> 
> Is this blocking any PRs or issues right now? If so, what's being blocked?

yes, it isn't possible to plug in the `Block` trait into `reth-node-types`, if the types in the signatures of the trait methods of `Block` aren't supported, ref https://github.com/paradigmxyz/reth/pull/11399

File: Makefile
Comment (emhane) [inline Makefile:1]: will revert

File: crates/primitives/src/block.rs (hunk 2 above)
Comment (emhane) [reply crates/primitives/src/block.rs:302]: `SealedBlock` is a return type in signature of `Block` trait

**Pull request closed without merging**